src/python/py/models/builder.py

```diff
@@ -291,6 +291,9 @@ def make_genai_config(self, model_name_or_path, extra_kwargs, out_dir):
                 "pad_token_id": config.pad_token_id if hasattr(config, "pad_token_id") and config.pad_token_id is not None else config.eos_token_id[0] if isinstance(config.eos_token_id, list) else config.eos_token_id,
                 "type": self.model_type[ : self.model_type.find("For")].lower(),
                 "vocab_size": self.vocab_size,
+                "model_name": model_name_or_path,
+                "precision": precision,
+                "execution_provider" :execution_provider
```

Comment:
The `execution_provider` can be obtained from [here](https://github.com/microsoft/onnxruntime-genai/blob/170145d7aaf6c88eeb5292c465b97e7941428e22/src/python/py/models/builder.py#L275) using [this information](https://github.com/microsoft/onnxruntime-genai/blob/170145d7aaf6c88eeb5292c465b97e7941428e22/src/python/py/models/builder.py#L316-L318) so we don't need a separate entry in the config for it.